src/FauxControl/FauxControl.js

```diff
@@ -0,0 +1,430 @@
+    };
+  },
+  underlay: ({ disabled, readOnly, theme: baseTheme, variant }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      backgroundColor:
+        disabled || readOnly
+          ? theme.backgroundColor_disabled
+          : theme.FauxControl_backgroundColor,
+      borderColor:
+        variant && !disabled && !readOnly
+          ? theme[`borderColor_${variant}`]
+          : theme.FauxControl_borderColor,
+      borderRadius: theme.FauxControl_borderRadius,
+      borderStyle: 'solid',
+      borderWidth: theme.FauxControl_borderWidth,
+      bottom: 0,
+      left: 0,
+      position: 'absolute',
+      right: 0,
+      top: 0,
+      zIndex: -1
+    };
+  }
+};
+
+const Prefix = createStyledComponent('span', styles.prefix);
+const Root = createStyledComponent('div', styles.root, {
+  displayName: 'FauxControl',
+  includeStyleReset: true
+});
+const Suffix = createStyledComponent('span', styles.suffix);
+const Underlay = createStyledComponent('div', styles.underlay, {
+  displayName: 'Underlay'
+});
+
+const variantIcons = {
+  danger: <IconDanger />,
+  success: <IconSuccess />,
+  warning: <IconWarning />
+};
+
+function getIcons({
+  disabled,
+  iconStart,
+  iconEnd,
+  readOnly,
+  size,
+  variant,
+  variantIcons
+}) {
+  if (disabled || readOnly) {
+    return [];
+  }
+
+  const iconSize = size === 'small' ? 'medium' : pxToEm(24);
+  const startIcon =
+    iconStart &&
+    cloneElement(iconStart, {
+      size: iconSize,
+      key: 'iconStart'
+    });
+
+  const endIconSource = variant
+    ? variantIcons[variant]
+    : iconEnd ? iconEnd : null;
+
+  const endIcon =
+    endIconSource &&
+    cloneElement(endIconSource, {
+      size: iconSize,
+      key: 'iconEnd'
+    });
+
+  return [startIcon, endIcon];
+}
+
+/**
+ * FauxControl
+ */
+export default class FauxControl extends Component<Props> {
+  render() {
+    const {
+      afterItems,
+      beforeItems,
+      children,
+      control,
+      controlProps: controlPropsIn,
+      disabled,
+      fauxControlRef,
+      iconEnd,
+      iconStart,
+      prefix: prefixIn,
+      size,
+      readOnly,
+      suffix: suffixIn,
+      variant,
+      ...restProps
+    } = this.props;
+
+    const rootProps = {
+      disabled,
+      innerRef: fauxControlRef,
+      variant,
+      ...restProps
+    };
+
+    const [startIcon, endIcon] = getIcons({
+      disabled,
+      iconStart,
+      iconEnd,
+      readOnly,
+      size,
+      variant,
+      variantIcons
+    });
+
+    const prefixAndSuffixProps = {
+      iconEnd,
+      iconStart,
+      size,
+      variant
+    };
+
+    const prefix = prefixIn ? (
+      <Prefix {...prefixAndSuffixProps} key="prefix">
+        {prefixIn}
+      </Prefix>
+    ) : null;
+    const suffix = suffixIn ? (
+      <Suffix {...prefixAndSuffixProps} key="suffix">
+        {suffixIn}
+      </Suffix>
+    ) : null;
+
+    const controlProps = {
+      controlPropsIn,
+      ...controlPropsIn,
```

Comment:
Do we need to have both of these?

Reply:
Yes.  The first is used in the application of styles and the latter is used to actually spread the props on the control.

The styles code needs `controlPropsIn.variant`, which gets overridden by `props.variant` in the definition of `controlProps`, which means that it would otherwise be unavailable to the styles code unless passed in specifically under another name.  In this case, we are just passing the entire `controlPropsIn` object for simplicity.  

There is a need to differentiate these as we may want to apply a variant inside the control, but not to the entire thing.

<img width="399" alt="screen shot 2018-03-09 at 6 40 43 am" src="https://user-images.githubusercontent.com/202773/37210170-fec02ce8-2364-11e8-8b1b-fd3101a39165.png">